rust/lance/src/io/commit/conflict_resolver.rs

```diff
@@ -0,0 +1,722 @@
+                    and it was missing transaction metadata.",
+                    other_version
+                ),
+                location: location!(),
+            });
+        };
+
+        match self.transaction.conflicts_with(other_transaction) {
+            ConflictResult::Compatible => Ok(()),
+            ConflictResult::NotCompatible => {
+                Err(crate::Error::CommitConflict {
+                    version: other_version,
+                    source: format!(
+                        "This {} transaction is incompatible with concurrent transaction {} at version {}.",
+                        self.transaction.operation, other_transaction.operation, other_version).into(),
+                    location: location!(),
+                })
+            },
+            ConflictResult::Retryable => {
+                match &other_transaction.operation {
+                    Operation::Update { updated_fragments, removed_fragment_ids, .. } |
+                    Operation::Delete { updated_fragments, deleted_fragment_ids: removed_fragment_ids, .. } => {
+                        if self.affected_rows.is_none() {
+                            // We don't have any affected rows, so we can't
+                            // do the rebase anyways.
+                            return Err(self.retryable_conflict_err(
+                                other_transaction,
+                                other_version,
+                                location!()
+                            ));
+                        }
+                        for updated in updated_fragments {
+                            if let Some((fragment, needs_rewrite)) = self.initial_fragments.get_mut(&updated.id) {
+                                // If data files, not just deletion files, are modified,
+                                // then we can't rebase.
+                                if fragment.files != updated.files {
+                                    return Err(self.retryable_conflict_err(
+                                        other_transaction,
+                                        other_version,
+                                        location!()
+                                    ));
+                                }
+
+                                // Mark any modified fragments as needing a rewrite.
+                                *needs_rewrite |= updated.deletion_file != fragment.deletion_file;
+                            }
+                        }
+
+                        for removed_fragment_id in removed_fragment_ids {
+                            if self.initial_fragments.contains_key(removed_fragment_id) {
+                                return Err(self.retryable_conflict_err(
+                                        other_transaction,
+                                        other_version,
+                                        location!()
+                                    ));
+                            }
```

Comment:
If self is a delete operation, deleting some rows in a fragment, and other is a delete operation, deleting the entire fragment, then it seems like we don't need a conflict?

However, if self is an update operation, updating some rows in a fragment, and other is a delete operation, deleting the entire fragment, then I agree it seems like we need a conflict.

Same as above comment.  Probably best to just leave it as it is for simplicity even if we could maybe avoid a conflict in one or two rare scenarios.